### PR TITLE
Remove legacy type service usage from modern Open MCT codebase

### PR DESCRIPTION
**Pull request by akhenry**
Partially addresses https://github.com/nasa/openmct/issues/4535

Replaces usage of the Type Service in `DefaultMetadataProvider`. The remaining usage of the legacy type service is in the Create Button and Action, which will be addressed by https://github.com/nasa/openmct/pull/4478

Moves registration of the Folder object type and the Unknown Type out of a bundle and into a plugin.

Note that the ExportAsJSON spec still mocks out the TypeService in some of its disabled tests. [A followup issue](https://github.com/nasa/openmct/issues/4558) has been created to address this.

### All Submissions:

* [X] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [X] Changes address original issue?
* [X] Unit tests included and/or updated with changes?
* [X] Command line build passes?
* [X] Has this been smoke tested?
* [X] Testing instructions included in associated issue?

### CODE REVIEW

**Comment by akhenry** [comment]
Switched back to draft because forms refactor has now been merged.

File: src/api/telemetry/TelemetryAPISpec.js [hunk 1]
```diff
@@ -37,12 +36,12 @@ describe('Telemetry API', function () {
             ]),
             $injector: jasmine.createSpyObj('injector', [
                 'get'
```
Comment (jvigliotta) [inline src/api/telemetry/TelemetryAPISpec.js:38]: Do we still need injector?

**Review by jvigliotta — COMMENTED** [review]
Just got a question, otherwise this looks good!

File: src/api/telemetry/TelemetryAPISpec.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (akhenry) [reply src/api/telemetry/TelemetryAPISpec.js:38]: Don't think so, good catch! Thanks

[the remark this answers is not in the document]
Comment (akhenry) [reply src/api/telemetry/TelemetryAPISpec.js]: Fixed, thanks

**Review by jvigliotta — APPROVED** [review]
### Author Checklist
Check | Passed
-- | --
Changes address original issue? | Y
Unit tests included and/or updated with changes? | Y
Command line build passes? | Y
Changes have been smoke-tested? | Y
Testing instructions included? | Y

**Pull request merged**